### DISCUSSION TITLE
Relationships

### DISCUSSION THREAD

**Discussion by Kyzegs** [cjmellor/approval#42] [Ideas]
The package is great but I am missing support for relations. Whenever I have a model with the `MustBeApproved` trait I expect that I would be able to create relational entries that are attached to a certain approval.

```php
$convention = Convention::create($request->validated());
$convention->categories()->saveMany($categories);
$convention->dates()->createMany($request->get('dates'));
```

The `Convention` model in this case has the `MustBeApproved` trait. I expect the relational entries to be attached to the approval for the parent. Whenever the parent is approved then everything should be created at once.

**Comment by cjmellor** [comment]
Hi @Kyzegs 

It's a good idea suggestion, and not something I'd considered.

I'm open to a PR for this, as I aren't sure of my availability to look at it myself just yet.

Thanks!

**Comment by cjmellor** [comment]
Probably not related to your suggestion, but I found a bug (#43) after looking into your idea, so, thanks for that I guess 😂

**Comment by aymanalhattami** [comment]
@Kyzegs 
Did you find any solutions on how to implement this idea?

**Comment by Temepest74** [comment]
Any updates / solutions?

**Comment by cjmellor** [comment]
Not something I plan on working on any time soon, but still open to a PR for it
